### PR TITLE
Bug 1880285: hack/log-explainer: Render CVO logs for easier analysis

### PR DESCRIPTION
**Pull request by wking**
Make it easier to dig into the individual sync cycles, find goroutines that have hung, etc.

### CODE REVIEW

**Review by vrutkovs — COMMENTED** [review]
Looks great, few nits

**Comment by vrutkovs** [comment]
Running this on latest successful gcp upgrade:
```
curl -s https://storage.googleapis.com/origin-ci-test/logs/release-openshift-origin-installer-e2e-gcp-upgrade/1303960118940930048/artifacts/e2e-gcp-upgrade/pods/openshift-cluster-version_cluster-version-operator-6bb4c967bb-gj5dk_cluster-version-operator.log | ./log-explainer.py
```

This produces a lot of unclassified status lines:
```
2020-09-10 09:07:11.834840 status.go Synchronizing errs=field.ErrorList{} status=&cvo.SyncWorkerStatus{Generation:2, Step:"", Failure:error(nil), Fraction:1, Completed:3, Reconciling:true, Initial:false, VersionHash:"ZSdny7Ojtgc=", LastProgress:time.Time{wall:0xbfce99029b78f107, ext:868087265759, loc:(*time.Location)(0x28ced60)}, Actual:v1.Release{Version:"4.6.0-0.ci-2020-09-10-073144", Image:"registry.svc.ci.openshift.org/ocp/release:4.6.0-0.ci-2020-09-10-073144", URL:"", Channels:[]string(nil)}, Verified:false}
2020-09-10 09:07:11.834909 status.go merge into existing history completed=true desired=v1.Release{Version:"4.6.0-0.ci-2020-09-10-073144", Image:"registry.svc.ci.openshift.org/ocp/release:4.6.0-0.ci-2020-09-10-073144", URL:"", Channels:[]string(nil)} last=&v1.UpdateHistory{State:"Completed", StartedTime:v1.Time{Time:time.Time{wall:0x0, ext:63735322961, loc:(*time.Location)(0x28ced60)}}, CompletionTime:(*v1.Time)(0xc001775e40), Version:"4.6.0-0.ci-2020-09-10-073144", Image:"registry.svc.ci.openshift.org/ocp/release:4.6.0-0.ci-2020-09-10-073144", Verified:false}
```
These should get their own section I guess

File: hack/log-explainer.py [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+        'sync worker': [line for line in lines if line['file'] in ['sync_worker.go', 'task.go', 'task_graph.go', 'warnings.go']],
+        'upgradeable': [line for line in lines if line['file'] == 'upgradeable.go' or 'syncing upgradeable' in line['message']],
+    }
+
+    metadata['leader_elections'] = [line for line in lines if line['file'] == 'leaderelection.go']
+    acquired_leader_lease = [line for line in metadata['leader_elections'] if 'successfully acquired lease' in line['message']]
+    if acquired_leader_lease:
+        metadata['time_to_leader_lease'] = acquired_leader_lease[0]['time'] - metadata['start_time']
+
+    sync_worker = []
```
[the remark this answers is not in the document]
Comment (wking) [reply hack/log-explainer.py]: Removed in cce52b5969 -> 8410557dc2.

File: hack/log-explainer.py [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+                ]:
+            f.write('   <h3 id="{0}">{0}</h3>\n'.format(key))
+            gorountine_lines = metadata.get(key)
+            if gorountine_lines:
+                time_to_start = gorountine_lines[0]['time'] - metadata['start_time']
+                time_since_final_log = metadata['stop_time'] - gorountine_lines[-1]['time']
+                warning = ''
+                if (
+                        time_to_start.total_seconds() > 2*60 or
+                        (time_since_final_log.total_seconds() > 5*60 and key is not 'metrics')  # metrics server does not log periodically
```
[the remark this answers is not in the document]
Comment (wking) [reply hack/log-explainer.py]: Fixed in cce52b5969 -> 8410557dc2.

**Review by vrutkovs — APPROVED** [review]
/lgtm

**Comment by vrutkovs** [comment]
/retitle Bug 1880285: hack/log-explainer: Render CVO logs for easier analysis

**Pull request merged**